tests/legacy-cli/e2e/tests/i18n/ivy-localize-serviceworker.ts

```diff
@@ -0,0 +1,189 @@
+import { ng, npm } from '../../utils/process';
+import { updateJsonFile } from '../../utils/project';
+import { expectToFail } from '../../utils/utils';
+import { readNgVersion } from '../../utils/version';
+
+export default async function() {
+  let localizeVersion = '@angular/localize@' + readNgVersion();
+  if (getGlobalVariable('argv')['ng-snapshots']) {
+    localizeVersion = require('../../ng-snapshot/package.json').dependencies['@angular/localize'];
+  }
+  await npm('install', `${localizeVersion}`);
+
+  let serviceWorkerVersion = '@angular/service-worker@' + readNgVersion();
+  if (getGlobalVariable('argv')['ng-snapshots']) {
+    serviceWorkerVersion = require('../../ng-snapshot/package.json').dependencies[
+      '@angular/service-worker'
+    ];
+  }
+  await npm('install', `${serviceWorkerVersion}`);
+
+  await updateJsonFile('tsconfig.json', config => {
+    config.compilerOptions.target = 'es2015';
+    config.angularCompilerOptions.disableTypeScriptVersionCheck = true;
+  });
+
+  const baseDir = 'dist/test-project';
+
+  // Set configurations for each locale.
+  const langTranslations = [
+    { lang: 'en-US', translation: 'Hello i18n!' },
+    { lang: 'fr', translation: 'Bonjour i18n!' },
+  ];
+
+  await updateJsonFile('angular.json', workspaceJson => {
+    const appProject = workspaceJson.projects['test-project'];
+    const appArchitect = appProject.architect || appProject.targets;
+    const serveConfigs = appArchitect['serve'].configurations;
+    const e2eConfigs = appArchitect['e2e'].configurations;
+
+    // Make default builds prod.
+    appArchitect['build'].options.optimization = true;
+    appArchitect['build'].options.buildOptimizer = true;
+    appArchitect['build'].options.aot = true;
+    appArchitect['build'].options.fileReplacements = [
+      {
+        replace: 'src/environments/environment.ts',
+        with: 'src/environments/environment.prod.ts',
+      },
+    ];
+
+    // Enable service worker
+    appArchitect['build'].options.serviceWorker = true;
+
+    // Enable localization for all locales
+    appArchitect['build'].options.localize = true;
+
+    // Add locale definitions to the project
+    // tslint:disable-next-line: no-any
+    const i18n: Record<string, any> = (appProject.i18n = { locales: {} });
+    for (const { lang } of langTranslations) {
+      if (lang == 'en-US') {
+        i18n.sourceLocale = lang;
+      } else {
+        i18n.locales[lang] = `src/locale/messages.${lang}.xlf`;
+      }
+      serveConfigs[lang] = { browserTarget: `test-project:build:${lang}` };
+      e2eConfigs[lang] = {
+        specs: [`./src/app.${lang}.e2e-spec.ts`],
+        devServerTarget: `test-project:serve:${lang}`,
+      };
+    }
+  });
+
+  // Add service worker source configuration
+  const manifest = {
+    index: '/index.html',
+    assetGroups: [
+      {
+        name: 'app',
+        installMode: 'prefetch',
+        resources: {
+          files: [
+            '/favicon.ico',
```

Comment:
The files here don't seem right. 

They should be
```
          "/favicon.ico",
          "/index.html",
          "/manifest.webmanifest",
          "/*.css",
          "/*.js"
```

Reply:
Copied it directly from here: https://github.com/angular/angular-cli/blob/master/packages/angular_devkit/build_angular/test/browser/service-worker_spec_large.ts#L13